FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.m

```diff
@@ -232,4 +224,56 @@ - (instancetype)initWithGoogleAppID:(NSString *)appID
       });
 }
 
+#pragma mark - Delete FID
+
+- (FBLPromise<NSNull *> *)deleteInstallation {
+  return [self.deleteInstallationPromiseCache getExistingPendingOrCreateNewPromise];
+}
+
+- (FBLPromise<NSNull *> *)createDeleteInstallationPromise {
+  // Check for ongoing requests first, if there is no a request, then check local storage for
+  // existing instlallation.
+  FBLPromise<FIRInstallationsItem *> *currentInstallationPromise =
+      [self.authTokenForcingRefreshPromiseCache getExistingPendingPromise]
+          ?: [self.authTokenPromiseCache getExistingPendingPromise]
+                 ?: [self.getInstallationPromiseCache getExistingPendingPromise]
+                        ?: [self getStoredInstallation];
+
+  return currentInstallationPromise
+      .then(^id(FIRInstallationsItem *installation) {
+        return [self sendDeleteInstallationRequestIfNeeded:installation];
+      })
+      .then(^id(FIRInstallationsItem *installation) {
+        // Remove the installation from the local storage.
```

Comment:
Sorry for bring up the local cache thing again, but I think at some point when we have local cache, we might want to reset it as the first thing so that there's no race conditions. Blocking call ensures local cache is set and read immediately.

Reply:
No problem :) It work on it soon.